pcsx2-qt/Debugger/DisassemblyWidget.cpp

```diff
@@ -0,0 +1,699 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2022  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+
+#include "DisassemblyWidget.h"
+
+#include "DebugTools/DebugInterface.h"
+#include "DebugTools/DisassemblyManager.h"
+#include "DebugTools/Breakpoints.h"
+#include "DebugTools/MipsAssembler.h"
+
+#include "QtUtils.h"
+#include "QtHost.h"
+#include <QtGui/QMouseEvent>
+#include <QtWidgets/QMenu>
+#include <QtGui/QClipboard>
+#include <QtWidgets/QInputDialog>
+#include <QtWidgets/QMessageBox>
+
+using namespace QtUtils;
+
+DisassemblyWidget::DisassemblyWidget(QWidget* parent)
+	: QWidget(parent)
+{
+	ui.setupUi(this);
+
+	CreateCustomContextMenu();
+	connect(this, SIGNAL(customContextMenuRequested(QPoint)), SLOT(customMenuRequested(QPoint)));
+}
+
+DisassemblyWidget::~DisassemblyWidget() = default;
+
+void DisassemblyWidget::CreateCustomContextMenu()
+{
+	if (m_contextMenu)
+		return; // ???
+	m_contextMenu = new QMenu(this);
+
+	QAction* action = 0;
+	m_contextMenu->addAction(action = new QAction("Copy Address", this));
+	connect(action, SIGNAL(triggered()), SLOT(contextCopyAddress()));
```

Comment:
Don't need to use named signals/slots here, you can pass the function pointers in for Qt6 (also helps solve the multiple overload problem, although that's not the case here):
```cpp
connect(action, &QAction::triggered, this, &DisassemblyWidget::contextCopyAddress);
```